modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala

```diff
@@ -90,7 +91,16 @@ object CompareContentsResponseBuilder extends StrictLogging {
           key -> new MatchingRuleGroup(
             rules.rule.flatMap { matchingRule =>
               matchingRule.values.map { struct =>
-                MatchingRule.fromJson(PactCoreUtils.structToJson(toJavaProto(struct)))
+                val json = PactCoreUtils.structToJson(toJavaProto(struct))
+                if (json.size() > 0 || matchingRule.`type`.isEmpty) {
+                  MatchingRule.fromJson(json)
+                } else {
+                  MatchingRule.fromJson(
+                    Json.toJson(
+                      Map("match" -> matchingRule.`type`).asJava
+                    )
+                  )
+                }
```

Comment:
I was thinking sth like
```suggestion
                                MatchingRule.fromJson(
                  Option(PactCoreUtils.structToJson(toJavaProto(struct)))
                    .filterNot(ruleJson => ruleJson.size() == 0 && matchingRule.`type`.nonEmpty) match {
                    case Some(value) => value
                    case None => Json.toJson(Map("match" -> matchingRule.`type`).asJava)
                  }
                )
```

Reply:
I applied the suggested changes and readjusted the formatting to allow it to pass the`scalafmtCheckAll´.